pkg/integration/zsh.go

```diff
@@ -0,0 +1,6 @@
+package integration
+
+var zshSource = `
+prmptcmd() { $("__dad_prompt_command") }
+precmd_functions=(prmptcmd)
```

Comment:
`precmd_functions+=(prmptcmd)`
To preserve the other pre-command hooks.

Reply:
It could also be wrapped in a if statement in case dad is already activated
```
if [[ ! "${precmd_functions[@]}" == *__dad_prompt_command* ]]; then
  precmd_functions+=(prmptcmd)
fi
```